Kudu.Services/Deployment/PushDeploymentController.cs

```diff
@@ -171,33 +337,37 @@ private string GetZipURLFromJSON(JObject requestObject)
                     {
                         throw new ArgumentException("Invalid Url in the JSON request");
                     }
+
                     return packageUri;
                 }
                 catch (Exception ex)
                 {
                     _tracer.TraceError(ex, "Error reading the URL from the JSON {0}", requestObject.ToString());
+
                     throw;
                 }
             }
         }
 
-        private async Task<HttpResponseMessage> PushDeployAsync(ZipDeploymentInfo deploymentInfo, bool isAsync)
+        private async Task<HttpResponseMessage> PushDeployAsync(ArtifactDeploymentInfo deploymentInfo, bool isAsync)
         {
             var content = Request.Content;
+            var deployer = deploymentInfo.Deployer;
+            var deploymentType = Request.RequestUri.ParseQueryString()["type"];
             var isRequestJSON = content.Headers?.ContentType?.MediaType?.Equals("application/json", StringComparison.OrdinalIgnoreCase);
             if (isRequestJSON == true)
             {
                 try
                 {
                     var requestObject = await Request.Content.ReadAsAsync<JObject>();
-                    deploymentInfo.ZipURL = ArmUtils.IsArmRequest(Request) ? GetZipURLFromARMJSON(requestObject) : GetZipURLFromJSON(requestObject);
+                    deploymentInfo.ArtifactURL = ArmUtils.IsArmRequest(Request) ? GetArticfactURLFromARMJSON(requestObject) : GetArtifactURLFromJSON(requestObject);
                 }
                 catch (Exception ex)
                 {
                     return ArmUtils.CreateErrorResponse(Request, HttpStatusCode.BadRequest, ex);
                 }
             }
-            else
+            else if (deployer != Constants.OneDeploy || deploymentType == Constants.Zip) // only runs this code if using Zip/War Deploy or OneDeploy Zip 
```

Comment:
It would be good to add a comment that clearly explains why this needs to be done only for zip files and not for non-zip files